### PR TITLE
change rye to uv, add isort to ruff linter and pre-commit checks

### PR DESCRIPTION
**Pull request by EllieKallmier**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,12 +43,12 @@ If you are proposing a feature:
 Ready to contribute? Here's how to set up `package_name` for local development.
 
 1. Download a copy of `package_name` locally.
-2. Install [`rye`](https://rye.astral.sh/).
-3. Install `package_name` using `rye` by running `rye sync` in the project directory.
-4. Install the `pre-commit` git hook scripts that `package_name` uses by running the following code using `rye`:
+2. Install [`uv`](https://github.com/astral-sh/uv).
+3. Install `isp-workbook-parser` using `uv` by running `uv sync` in the project directory.
+4. Install the `pre-commit` git hook scripts that `isp-workbook-parser` uses by running the following code using `uv`:
```
Comment (prakaa) [inline CONTRIBUTING.md:48]: Change package name to `isp-trace-parser`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,12 +43,12 @@ If you are proposing a feature:
 Ready to contribute? Here's how to set up `package_name` for local development.
 
 1. Download a copy of `package_name` locally.
```
Comment (prakaa) [inline CONTRIBUTING.md:45]: Change package name to `isp_trace_parser` (check this across the entire package)

**Review by prakaa — CHANGES_REQUESTED** [review]
@EllieKallmier see comments

**Comment by prakaa** [comment]
The `isort` call as a part of `ruff` appears to be working given import order in `src/*.py` files is being changed

File: CONTRIBUTING.md (hunk 1 above)
Comment (EllieKallmier) [reply CONTRIBUTING.md]: done!

File: CONTRIBUTING.md (hunk 2 above)
Comment (EllieKallmier) [reply CONTRIBUTING.md]: done too - but used isp-trace-parser (not underscores) to be consistent - lmk if this is incorrect and I'll fix again :)

**Review by prakaa — APPROVED** [review]

**Pull request merged**